addons/udes_priorities/models/priority.py

```diff
@@ -26,17 +26,35 @@ class UdesPriority(models.Model):
         default=lambda self: uuid4(),
     )
 
+    @api.multi
+    @api.depends("reference")
+    def _compute_outstanding_pickings(self):
+        Picking = self.env["stock.picking"]
+        self.outstanding_pickings = Picking.search_count(
```

Comment:
We might still want to keep `self.ensure_one()`.

Reply:
I am still learning about when it is appropriate to use `ensure.one()` and because of that I'd like to ask, what would be the purpose of having this called here in a compute method? In which occasion would this be a problem? I assume that if the method is only used by a compute field then there is no need for calling `ensure_one()`?

Reply:
In this case it might not be needed, but in general @api.multi means we can have more than one record in self, which means if we do self.outstanding_pickings = XXX (when self contains more than one record) it will fail, instead we should use self.write(YYY). Usually we add a loop just in case:
```
for record in self:
      record.outstanding_pickings = XXX
```
At the end _compute_outstanding_pickings() is a function and can be called somewhere else (it shouldn't though).